### ISSUE TITLE
Remove mandatory ID command during session start

### ISSUE THREAD

**Issue by chaliy** [postalsys/imapflow#80]
**Describe the bug**

It seems like ID command is not part of the IMAP standard and some of the servers does not support it. In my case we use  dovecot.

When ID command is not supported on server, imapflow library outputs error message into the logs.

**To Reproduce**
```
await client.connect(); // <- here
let mailbox = await client.mailboxOpen('INBOX');
```

**Expected behavior**
No error message for "normal" behaviour

**Screenshots**

> {"level":20,"time":1658269323750,"pid":53836,"hostname":"FHJ2106FR4","component":"imap-connection","cid":"12juu9re9bmwqep8hcu1","src":"s","msg":"2 ID (\"name\" \"imapflow\" \"version\" \"1.0.100\" \"vendor\" \"Postal Systems\" \"support-url\" \"https://github.com/postalsys/imapflow/issues\")","cid":"12juu9re9bmwqep8hcu1"}
> {"level":20,"time":1658269323756,"pid":53836,"hostname":"FHJ2106FR4","component":"imap-connection","cid":"12juu9re9bmwqep8hcu1","src":"s","msg":"2 BAD invalid command","cid":"12juu9re9bmwqep8hcu1"}
> {"level":40,"time":1658269323756,"pid":53836,"hostname":"FHJ2106FR4","component":"imap-connection","cid":"12juu9re9bmwqep8hcu1","err":{"type":"Error","message":"Command failed","stack":"Error: Command failed\n    at ImapFlow.reader (/Users/mchalyi/Projects/uu3jrd/file-messenger/.webpack/main/index.js:33680:35)\n    at process.processTicksAndRejections (node:internal/process/task_queues:96:5)","response":{"tag":"2","command":"BAD","attributes":[{"type":"TEXT","value":"invalid command"}]},"responseStatus":"BAD","responseText":"invalid command"},"cid":"12juu9re9bmwqep8hcu1","msg":"Command failed"}

**Comment by chaliy** [comment]
If this is indeed confirmed as an issue, would you accept PR on making ID command optional?

**Comment by chaliy** [comment]
Okay, little bit more investigation. `dovecot` supports ID command.

> {"level":20,"time":1658269323742,"pid":53836,"hostname":"FHJ2106FR4","component":"imap-connection","cid":"12juu9re9bmwqep8hcu1","src":"s","msg":"1 CAPABILITY","cid":"12juu9re9bmwqep8hcu1"}
> {"level":20,"time":1658269323750,"pid":53836,"hostname":"FHJ2106FR4","component":"imap-connection","cid":"12juu9re9bmwqep8hcu1","src":"s","msg":"* CAPABILITY IMAP4 IMAP4rev1 UIDPLUS SASL-IR LOGIN-REFERRALS ID ENABLE IDLE LITERAL+ AUTH=PLAIN","cid":"12juu9re9bmwqep8hcu1"}

At the end of the day, it looks like `dovecot` supports ID command, however only after `LOGIN` command. My best guess, they made it to prevent unauthentificated connections to spam logs via ID command.

**Comment by andris9** [comment]
Imapflow handles ID correctly. Do you happen to have some kind of weird setup with a proxy, maybe, for example Nginx in front of Dovecot? What are the version numbers of Docecot and the proxy server?
